File: CodeChef/Practice/Medium/MARBLES.py
def nCr(n,r):
    if n-r < r:
        r = n-r
    answer = 1
    for i in range(r):
        answer*=(n-i)
    for i in range(1,r+1):
        answer/=i
    return answer

for _ in range(input()):
    n,k = map(int,raw_input().split())
    print nCr(n-1,k-1)

File: CodeChef/Practice/Medium/MARBLES.cpp
#include<iostream>
using namespace std;

long long nCr(int n, int r) {
  long long answer = 1;
  for (int i = 0; i < r; i++) {
    answer*=(n-i);
  }
  for (int i = 1; i <= r; i++) {
    answer/=i;
  }
  return answer;
}

int main() {
  int t;
  cin>>t;
  for (int i = 0; i < t; i++) {
    int n,k;
    cin>>n>>k;
    cout<<nCr(n-1,k-1)<<'\n';
  }
}
